morpheus/_lib/include/morpheus/stages/preallocate.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "morpheus/messages/meta.hpp"
+#include "morpheus/messages/multi.hpp"
+#include "morpheus/utilities/type_util_detail.hpp"  // for TypeId
+
+#include <pysrf/node.hpp>
+#include <rxcpp/rx.hpp>
+#include <srf/node/sink_properties.hpp>    // for SinkProperties<>::sink_type_t
+#include <srf/node/source_properties.hpp>  // for SourceProperties<>::source_type_t
+#include <srf/segment/builder.hpp>
+#include <srf/segment/object.hpp>  // for Object
+
+#include <exception>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace {
+/**
+ * @brief Performs preallocation to the underlying dataframe. These functions ensure that the MutableTableInfo object
+ * has gone out of scope and thus releasing the mutex prior to the stage calling `on_next` which may block.
+ *
+ * @param msg
+ * @param column_names
+ * @param column_types
+ */
+void preallocate(std::shared_ptr<morpheus::MessageMeta> msg,
+                 const std::vector<std::string> &column_names,
+                 const std::vector<morpheus::TypeId> &column_types)
```

Comment:
I went with that since that is what the `MutableTableInfo.insert_missing_columns` expected. Although I could easily update the `insert_missing_columns` method as well.